featurebyte/models/warehouse_table.py

```diff
@@ -0,0 +1,45 @@
+"""
+WarehouseTableModel
+"""
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Optional
+
+import pymongo
+
+from featurebyte.models.base import FeatureByteCatalogBaseDocumentModel
+from featurebyte.query_graph.model.common_table import TabularSource
+
+
+class WarehouseTableModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    WarehouseTableModel class
+
+    Represents a catalog specific table in the warehouse with optional metadata.
+
+    tag is an optional string to identify a collection of tables, such as all temporary tile tables
+    created for the purpose of creating a historical feature table, in which case the tag could be
+    derived based on the historical feature table id.
+
+    expires_at is an optional datetime to indicate when the table should be deleted for clean up
+    purpose.
+    """
+
+    location: TabularSource
+    tag: Optional[str] = None
+    expires_at: Optional[datetime] = None
```

Comment:
Should we add a `warehouse_tables` property attribute like https://github.com/featurebyte/featurebyte/blob/0e8c5032e36e9014b51edb0d6a02f2f1f369bd73/featurebyte/models/tile_registry.py#L83-L85 so that the temporary table is removed during catalog cleanup?

Reply:
Sure, will add this property.